src/odemis/gui/cont/correlation.py

```diff
@@ -0,0 +1,269 @@
+class CorrelationController(object):
+
+    def __init__(self, tab_data, panel, tab, viewports):
+        """
+        tab_data (MicroscopyGUIData): the representation of the microscope GUI
+        panel (wx._windows.Panel): the panel containing the UI controls
+        tab: (Tab): the tab which should show the data
+        viewports: (ViewPorts): the tab view ports
+        """
+        
+        self._tab_data_model = tab_data
+        self._main_data_model = tab_data.main
+        self._panel = panel
+        self._tab = tab
+        self._viewports = viewports
+
+        # selected stream
+        self.selectedStream = model.VigilantAttribute(None)
+
+        # disable if no streams are present
+        self._tab_data_model.streams.subscribe(self._on_correlation_streams_change, init=True)
+        self._tab_data_model.streams.subscribe(self._update_correlation_cmb, init=True)
+
+        # connect the correlation streams to the tab data
+        self._panel.cmb_correlation_stream.Bind(wx.EVT_COMBOBOX, self._on_selected_stream_change)
+
+        for vp in self._viewports:
+            vp.canvas.Bind(wx.EVT_CHAR, self._on_char)
+            vp.canvas.Bind(wx.EVT_LEFT_DOWN, self._on_mouse_down)
+       
+    def _update_correlation_cmb(self, streams):
+
+        # keep the combobox in sync with streams
+        self._panel.cmb_correlation_stream.Clear()
+        for s in streams:
+            self._panel.cmb_correlation_stream.Append(s.name.value, s)
+        
+        # select the first stream, if available
+        if len(streams) > 0:
+            self._panel.cmb_correlation_stream.SetSelection(0)
+    
+    @call_in_wx_main
+    def _on_correlation_streams_change(self, streams):
+        """hide/show the correlation panel if there are no streams"""
+        visible = len(streams) != 0
+        self._panel.fp_meteor_correlation.Show(visible)
+
+    def _on_selected_stream_change(self, evt):
+        """change the selected stream to the one selected in the combo box"""
+        idx = self._panel.cmb_correlation_stream.GetSelection()
+        self.selectedStream.value = self._panel.cmb_correlation_stream.GetClientData(idx)
+        logging.info(f"Selected Stream Changed to {idx}: {self.selectedStream.value.name.value}")
+    
+    def _reset_stream_correlation_data(self, s: StaticStream):
+        """reset the stream position to the original position / rotation / scale"""
+        s.raw[0].metadata[model.MD_POS_COR] = (0, 0)
+        s.raw[0].metadata[model.MD_ROTATION_COR] = 0
+        s.raw[0].metadata[model.MD_PIXEL_SIZE_COR] = (1, 1)
+            
+    def _add_correlation_stream(self, s: StaticStream): 
+        """add a stream to the correlation controller"""
+        # reset the stream correlation data
+        self._reset_stream_correlation_data(s)
+
+        # add the stream to the correlation streams
+        self._tab_data_model.streams.value.append(s)
+        
+
+    def _update_correlation_streams(self, streams):
```

Comment:
It seems this function is the same as `_add_correlation_steam` but supports passing multiple streams. Is that correct, or are you going to change it to also remove streams?

In general, the docstring is missing so I'm not sure what your intention was, but I feel the name is not matching the behaviour. If it's "update" then probably it should also removes the streams which are not listed in the argument? Also why "correlation" ? Is that because it resets the correction metadata?

Reply:
changed to add_streams, added docstring and note explaining intended behaviour